contracts/MechMarketplace.sol

```diff
@@ -141,19 +150,37 @@ contract MechMarketplace is IErrorsMarketplace {
         );
     }
 
-    /// @dev Registers service as a mech.
-    /// @param serviceId Service id.
-    /// @param mechFactory Mech factory address.
-    /// @return mech The created mech instance address.
-    function create(uint256 serviceId, address mechFactory, bytes memory payload) external returns (address mech) {
-        mech = IMechFactory(mechFactory).createMech(address(this), serviceRegistry, serviceId, payload);
+    /// @dev Processes payment for request delivery.
+    /// @param deliveryMech Delivery agent mech address.
+    /// @param requestId Request id.
+    function _processPayment(address deliveryMech, uint256 requestId) internal virtual {
+        // Reentrancy guard
+        if (_locked > 1) {
+            revert ReentrancyGuard();
+        }
+        _locked = 2;
 
-        // This should never be the case
-        if (mech == address(0)) {
-            revert ZeroAddress();
+        // Get request Id payment
+        uint256 payment = mapRequestIdPayments[requestId];
+
+        // Process payment
+        if (payment > 0) {
+            uint256 fee = IMechManager(mechManager).fee();
```

Comment:
I don't want to have multiple marketplaces. Don't like the complexity this introduces. imo it's much simpler to scope fees logic to mechs. Then marketplace interface stays the same and simply the factory supports new types of mechs. A mech specifies its fee model that gets retrieved here.

Reply:
Refactoring... but fee needs further clarification